src/System.Private.CoreLib/shared/System/Threading/WaitHandle.cs

```diff
@@ -0,0 +1,384 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Threading
+{
+    public abstract partial class WaitHandle : MarshalByRefObject, IDisposable
+    {
+        internal const int WaitAbandoned = 0x80;
+        internal const int WaitTimeout = 0x102;
+        internal const int ErrorTooManyPosts = 0x12A;
```

Comment:
I generally agree, but it's tricky since they are reused on Unix builds both by CoreCLR and CoreRT. In case of CoreRT I believe the Windows Interop files are not included. I will check.

Reply:
Looks like I can remove ErrorTooManyPosts from the shared file and move the error handling to CoreCLR-specific part. As for the rest - I will probably wrap it up as an `enum` and put a comment on it.